client/src/components/ContestTable/ContestTable.tsx

```diff
@@ -12,152 +19,197 @@ import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
 import TableCell from '@material-ui/core/TableCell';
 import TableBody from '@material-ui/core/TableBody';
-import TablePagination from '@material-ui/core/TablePagination';
-import { getAllContests } from '../../helpers/APICalls/contest';
-import { Contest } from '../../interface/User';
+import SortIcon from '@material-ui/icons/Sort';
+import { Animated } from 'react-animated-css';
+import useStyles from './useStyles';
 import SectionHeader from '../../components/SectionHeader/SectionHeader';
 import { Link } from 'react-router-dom';
-import moment from 'moment';
-import useStyles from './useStyles';
-
 
 const ContestTable = (): JSX.Element => {
+  const [contests, setContests] = useState<Contest[]>([]);
+  const [numContests, setNumContests] = useState<number>(contests.length);
+  const [sortType, setSortType] = useState<keyof Contest>('deadlineDate');
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [dateFilter, setDateFilter] = useState<any>();
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
 
-    const classes = useStyles();
-    const [sortType, setSortType] = useState<keyof Contest>('deadlineDate');
-    const [page, setPage] = useState(0);
-    const [rowsPerPage, setRowsPerPage] = useState(10);
-    const [contests, setContests] = useState<Contest[]>([]);
-    const [dateFilter, setDateFilter] = useState<any>();
+  const getContestsLength = () => {
+    getNumContests().then((data: number) => {
```

Comment:
Why are we making an API call to get the number of contests here, can't we use the length of the contests array? :)

Reply:
because we need to know how many total active contests there are in total in the database when there is no filter. The contest state will only have 10, 25, etc. results due to the back end pagination only sending a batch of results at a time.

Reply:
While I do feel like we need this call to get this information, I moved it up one level so it only runs once when the page renders.

Reply:
Thanks for the clarification and also moving the API call up!